client/src/modules/user/AuthBox/components/LinkExpired.jsx

```diff
@@ -14,7 +16,24 @@ import { Routes } from 'common/constants/enums';
 class LinkExpired extends PureComponent {
   state = {
     isLinkSuccessfullySent: false,
-    sendingFailed: false
+    sendingFailed: false,
+    isTokenValid: false
+  };
+
+  componentDidMount() {
+    this.checkToken();
+  }
+
+  checkToken = () => {
+    const {
+      match: {
+        params: { token }
+      }
+    } = this.props;
+
+    checkToken(token)
```

Comment:
Token validation complicates this IMHO. Consider just returning a proper error message from the backend when consuming the token.